piksi_tools/console/output_list.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+# Copyright (C) 2011-2014 Swift Navigation Inc.
+# Contact: Fergus Noble <fergus@swift-nav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+"""
+Contains the class OutputList
+Displays Device Log messages and STDOUT/STDERR.
+"""
+
+from traits.api import HasTraits, Str, Bool, Trait, Int, List, Font, Float, \
+                       Enum, Property
+from traitsui.api import View, UItem, TabularEditor
+from pyface.api import GUI
+from traitsui.tabular_adapter import TabularAdapter
+import time
+
+# These levels are identical to sys.log levels
+LOG_EMERG      = 0       # system is unusable
+LOG_ALERT      = 1       # action must be taken immediately
+LOG_CRIT       = 2       # critical conditions
+LOG_ERROR      = 3       # error conditions
+LOG_WARN       = 4       # warning conditions
+LOG_NOTICE     = 5       # normal but significant condition
+LOG_INFO       = 6       # informational
+LOG_DEBUG      = 7       # debug-level messages
+
+# These log levels are defined uniquely to this module to handle the
+# list for stdout, stderror, and the device's log messages
+# An unknown log level will end up with - 2
+# A python stdout or stderr should come in as - 1
+
+CONSOLE_LOG_LEVEL = -1
+DEFAULT_LOG_LEVEL = -2
+
+# This maps the log level numbers to a human readable string
+# The unused log levels are commented out of the dict until used
+
+# OTHERLOG_LEVELS will be unmaskable
+OTHERLOG_LEVELS  = {
+                    CONSOLE_LOG_LEVEL: "CONSOLE",
+                    }
+# SYSLOG_LEVELS can be filtered
+
+SYSLOG_LEVELS = {#LOG_EMERG : "EMERG",
```

Comment:
For UI considerations.  I didn't want to have too many options in the drop-down filter.